src/wdio/services/ExpressDevServerService.js

```diff
@@ -0,0 +1,109 @@
+import express from 'express';
+import webpack from 'webpack';
+import MemoryFS from 'memory-fs';
+import mime from 'mime-types';
+import path from 'path';
+
+
+export default class WebDevServerService {
+  async onPrepare(config) {
+    if (!config.webpackConfig) {
+      // eslint-disable-next-line no-console
+      console.log('[WebDevService] No webpack configuration provided');
+      return;
+    }
+
+    const webpackConfig = config.webpackConfig;
+    const port = config.webpackPort || 8080;
+
+    await WebDevServerService.startExpressDevServer(webpackConfig, port).then((server) => {
+      this.server = server;
+    });
+  }
+
+  async onComplete() {
+    await this.stop();
+  }
+
+  static startExpressDevServer(webpackConfig, port) {
+    return WebDevServerService.compile(webpackConfig).then((fs) => {
+      const app = express();
+
+      // Setup a catch all route, we can't use 'static' because we need to use a vfs
+      app.get('*', (req, res, next) => {
+        let filename = req.url;
+        // Setup a default to route / to the index.html file.
+        // Should this be configurable in the future?
```

Comment:
Can we make this configurable now? We can access global webdriver configuration within this service. We can add/ the `express` key to pass object of information. I am thinking:

```
// wdio 
express: {
    defaultRoute: // defaults to '/index.hml'
}
```

Reply:
We could, but do we need to? index.html is what all our sites are set to.

Reply:
It would open it up for consumers use

Reply:
Fair enough, i'll look into it.

Reply:
I went with:
`//wdio
expressDevServer: {
    index: // defaults to '/index.hml'
}
`
I used index because that more closely aligns with the webpackdev server api. https://webpack.js.org/configuration/dev-server/#devserver-index